pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/DeprecatedAttribute.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.ast.xpath;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/**
+ * Node attribute getter methods might be annotated with {@code DeprecatedAttribute}
+ * to mark the attribute as deprecated for XPath. Unlike {@link Deprecated}, this
+ * annotation does not deprecate the method for java usage.
+ * 
+ * @since 6.21.0
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface DeprecatedAttribute {
```

Comment:
Maybe we should place this in an internal package? This may change until 7.0

Reply:
yes, I'll update the PR shortly